lib/linguist/languages.yml

```diff
@@ -4020,15 +4020,11 @@ M4Sugar:
 MATLAB:
   type: programming
   color: "#e16737"
-  aliases:
-  - octave
   extensions:
   - ".matlab"
   - ".m"
   tm_scope: source.matlab
   ace_mode: matlab
-  codemirror_mode: octave
-  codemirror_mime_type: text/x-octave
```

Comment:
Whilst Octave isn't Matlab, I think we should keep this for current Matlab users unless this is really is horribly wrong.

Reply:
Apart from providing a correct syntax highlight for Octave code, the other reason behind this PR is to segregate the two languages in order to prevent people from falsely assuming that MATLAB and Octave are bidirectionally compatible. GNU Octave can run MATLAB code, but MATLAB will not run Octave code. IMHO, keeping the `codemirror_mode` in MATLAB's definition is not a good idea.

Reply:
This isn't about running the code or giving an impression is would work, it's about getting syntax highlighting in the code editor box. Without this, users get nothing. With it, they'll get some syntax highlighting.

Reply:
I am not sure I follow your comment. Isn't the case that MATLAB code gets highlighted according to MATLAB's syntax tmLanguage file and Octave code (once merged and deployed in linguist) will get the syntax highlight from the new grammar tmLanguage file published in gnu-octave/Octave-language-grammar repo? In what case, the code editor will not apply any syntax highlighting? In any case, if you consider that the presence of codemirror code is a necessity for linguist's proper (or safe) operation, then I assume we can add it.

Reply:
> Isn't the case that MATLAB code gets highlighted according to MATLAB's syntax tmLanguage file and Octave code (once merged and deployed in linguist) will get the syntax highlight from the new grammar tmLanguage file published in gnu-octave/Octave-language-grammar repo?

Yes, when rendered once you've commited and pushed the files to GitHub. 

From the very top of `languages.yml`:

https://github.com/github-linguist/linguist/blob/39fd5e93de98de3434cb3e857c5b1972c418f8c5/lib/linguist/languages.yml#L12-L15

To extend on this, the codemirror options are used for the editor when you edit files directly on the GitHub website. Find a file in a repo on GitHub, open it and click the pencil icon - that syntax highlighting is coming from Codemirror, not the grammar included in Linguist.

We've not been consistent with ensuring we set these options in the past as not all languages have Codemirror support but your changes reminded me that this is a situation where it does.

If you remove this here, anyone who uses the GitHub UI to edit Matlab files will lose whatever syntax highlighting they get now, whether it's 100% accurate or not. 

This is also why I suggested adding it to the Octave language... that way peeps editing Octave files directly on GitHub will get syntax highlighting in the code editor. It seems silly not to offer it when it is available.

Reply:
Ah, ok, it totally makes sense. Thanks for clarifying what codemirror actually does. I will push these changes shortly.